app/models/katello/repository.rb

```diff
@@ -183,6 +183,8 @@ class Repository < Katello::Model
     scoped_search :on => :redhat, :complete_value => { :true => true, :false => false }, :ext_method => :search_by_redhat
     scoped_search :on => :container_repository_name, :complete_value => true
     scoped_search :on => :description, :only_explicit => true
+    scoped_search :on => :name, :relation => :product, :rename => :product_name
```

Comment:
Adding `product_name` here fixes a bug that you cannot search enabled repos by `product_name`.

Reply:
Acutally i realized that you can already search here by product name, we just rename it to 'product':

 scoped_search :rename => :product, :on => :name, :relation => :product, :complete_value => true

is that good enough, or do you actually need to search by product_name specifically?

Reply:
Yes I Do do actually need to search by `product_name` specifically!

Because the same search query should work for both `available` and `enabled` repos.